scripts/tests/test-ci-all.sh

```diff
@@ -237,7 +237,7 @@ fi
 if [ -n "${FM_TEST_CI_ALL_JOBS:-}" ]; then
   # when specifically set, use the env var
   parallel_args+=(--jobs "${FM_TEST_CI_ALL_JOBS}")
-elif [ -n "${CI:-}" ]; then
+elif [ -n "${CI:-}" ] || [ "${CARGO_PROFILE:-}" == "ci" ]; then
```

Comment:
This is the main reason. We were actually running CI with 3 jobs parallel max, while we wanted 8 (num cpus).

Reply:
Interesting, I would have expected `CI=true` from the GHA docs

https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

Reply:
That was the mistake. It runs in nix derivation so CI envvar is not set! 😂

Reply:
I probably assumed the same thing and didn't notice.

Reply:
Ah! Nice find 😄